thespian/system/transport/TCPTransport.py

```diff
@@ -340,13 +340,12 @@ def getAdminAddr(capabilities):
     @staticmethod
     def getAddressFromString(addrspec, adminRouting=False):
         if isinstance(addrspec, tuple):
-            addrparts = addrspec
+            addrparts = list(addrspec)
         else:
             addrparts = addrspec.split(':')
         addrtype = RoutedTCPv4ActorAddress if adminRouting else TCPv4ActorAddress
-        if 1 == len(addrparts):
-            return ActorAddress(addrtype(addrparts[0], DEFAULT_ADMIN_PORT, external=True))
-        return ActorAddress(addrtype(addrparts[0], addrparts[1], external=True))
+        ip, port = addrparts[0], addrparts[1] if len(attrparts) > 1 else DEFAULT_ADMIN_PORT
+        return ActorAddress(addrtype(ip, port, external=True))
```

Comment:
I was going more for clarity than performance here; trivial variable allocations are... trivial :)

If you want efficiency, how about:
```
addrparts = (addrparts + [ DEFAULT_ADMIN_PORT ])[:2]
return ActorAddress(addrtype(*addrparts, external=True))
```
?